crates/oxc_transformer/src/typescript/namespace.rs

```diff
@@ -150,9 +151,15 @@ impl<'a> TypeScript<'a> {
             &ctx.generate_uid_in_current_scope(real_name, SymbolFlags::FunctionScopedVariable),
         );
 
+        // TODO: This is workaround for now.
+        // Reuse TSModuleBlock's scope id in transformed function.
+        // We also need to update the scope flags to FunctionScopedVariable
+        let mut scope_id = None;
+
         let namespace_top_level = if let Some(body) = decl.body {
             match body {
                 TSModuleDeclarationBody::TSModuleBlock(mut block) => {
+                    scope_id = block.scope_id.get();
```

Comment:
Can we get the `scope_id` in the other branch for `TSModuleDeclarationBody::TSModuleDeclaration` as well?

Reply:
`TSModuleDeclarationBody::TSModuleDeclaration` has no scope id. You can see https://oxc-project.github.io/oxc/playground/?code=3YCAAIACgICAgICAgICXYPwo06VHxJ7qhgzVoEhzbUbnGCOcafvmPsPEMeUylof9IpeQ%2BId0nsXmmyubkBQxGJ1%2B8VzwpE5XnX5OFaO%2F721rUgVJ3398R6CA

Reply:
I meant can we use the `scope_id` from the `TSModuleBlock` within the inner `TSModuleDeclaration`:
<img width="1416" alt="namespace AST" src="https://github.com/oxc-project/oxc/assets/557937/3ddce694-2a31-475f-9d0e-77a4132e1337">

Reply:
Anyway, I'll follow-up, as discussed below.